h2o-hadoop/scripts/docker/build.py

```diff
@@ -0,0 +1,67 @@
+#! /usr/bin/python
+
+import argparse
+import subprocess
+import os
+import sys
+
+DISTRIBUTION_CDH = 'CDH'
+SUPPORTED_CDH_VERSIONS = ['5.4', '5.5', '5.6', '5.7', '5.8', '5.10']
+DISTRIBUTION_HDP = 'HDP'
+SUPPORTED_HDP_VERSIONS = ['2.2', '2.3', '2.4', '2.5', '2.6']
+SUPPORTED_DISTRIBUTIONS = [DISTRIBUTION_CDH, DISTRIBUTION_HDP]
+
+def pretty_print_array(array_to_print):
+    if array_to_print is None or len(array_to_print) == 0:
+        return ""
+    if len(array_to_print) == 1:
+        return str(array_to_print[0])
+    else:
+        return "%s and %s" % (", ".join(array_to_print[:-1]), array_to_print[-1])
+
+
+def init_args_parser():
+    supported_distributions_text = pretty_print_array(SUPPORTED_DISTRIBUTIONS)
+    parser = argparse.ArgumentParser(description='Builds a Dockerfile for given Hadoop distribution and version.')
+    parser.add_argument('-d', '--distribution',
+        type=str, choices=SUPPORTED_DISTRIBUTIONS,
+        help='Distribution of Hadoop. Currently %s are supported.' % supported_distributions_text,
+        required=True
+    )
+    parser.add_argument('-v', '--version', type=str,
+        help='Version of the Hadoop distribution.', required=True
+    )
+    parser.add_argument('-s', '--spark-version', choices=['1.6', '2.0', '2.1', '2.2'], action='append', help="Version of Spark which should be installed.")
```

Comment:
I need to be able to pass arbitrary spark version - eg. 2.2.1

Reply:
Added support for version format x.x.x